### ISSUE TITLE
open.spotify.com: ads / breakages

### ISSUE THREAD

**Issue by boromirmonk** [uBlockOrigin/uAssets#22198]
### Prerequisites

- [x] This is **NOT** a [_YouTube_](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [_Facebook_](https://www.github.com/uBlockOrigin/uAssets/issues/3367), [_Twitch_](https://www.github.com/uBlockOrigin/uAssets/issues/5184) or [_a shortener/hosting site_](https://github.com/uBlockOrigin/uAssets/discussions/17361) report. _These sites MUST be reported by clicking their **respective links**_.
- [x] I read and understand the [policy about what is a valid filter issue](https://github.com/uBlockOrigin/uAssets/blob/master/README.md#uassets).
- [x] I verified that this issue is not a duplicate. (Use this [button](https://user-images.githubusercontent.com/585534/146582579-c32707a0-36a1-4cc5-ad50-83172c9f67a8.png) to find out.). Comment in the old issue threads even when they are closed or even if you have a different problem.
- [x] I did not remove any of the [default filter lists](https://github.com/user-attachments/assets/1879fc88-09bc-4de4-9e2f-3834a6ab396a), or I have verified that the issue was not caused by removing any of the default lists.
- [x] I did not enable additional filter lists, or I have verified that the issue still occurs without enabling additional filter lists.
- [x] I do not have custom filters/rules, or I have verified that the issue still occurs without custom filters/rules.
- [x] I have verified that the web browser's built-in content blocker/tracking protection, [network wide/DNS blocking](https://en.wikipedia.org/wiki/DNS_blocking), or my VPN is not causing the issue.
- [x] I have turned off all other extensions and the issue still persists. (exception "[Firefox Multi-Account Containers](https://addons.mozilla.org/en-US/firefox/addon/multi-account-containers)").
- [x] If this is about a breakage or detection, I have verified that it is caused by uBlock Origin and isn't a site or browser issue.
- [ ] I did not answer truthfully to **ALL** the above checkboxes.
### URL address of the web page

`https://open.spotify.com/`

### Category

ads

### Description

Today, the site starting showing ads while having uBlock Origin activated. 

### Other extensions used

None.

### Screenshot(s)

<details>
<summary>Screenshot(s)</summary>

</details>

### Configuration

<details>

```yaml
uBlock Origin: 1.55.0
Chromium: 121
filterset (summary):
 network: 118987
 cosmetic: 51582
 scriptlet: 20570
 html: 0
listset (total-discarded, last-updated):
 default:
  user-filters: 0-0, never
  easylist: 79742-14, now
  easyprivacy: 43563-40, now
  plowe-0: 3783-1173, now
  spa-0: 3821-19, 16m
  spa-1: 5790-218, now
  ublock-badware: 7791-136, now
  ublock-filters: 36681-225, now
  ublock-privacy: 992-6, now
  ublock-quick-fixes: 128-5, now
  ublock-unbreak: 2192-34, now
  urlhaus-1: 8675-0, 16m
filterset (user): [empty]
trustedset:
 removed:
  chrome-scheme
switchRuleset:
 added: [array of 2 redacted]
userSettings: [none]
hiddenSettings: [none]
supportStats:
 allReadyAfter: 424 ms (selfie)
 maxAssetCacheWait: 202 ms
 cacheBackend: browser.storage.local
popupPanel:
 blocked: 8
 network:
  doubleclick.net: 3
  sentry.io: 5
```
</details>

**Comment by zaraford** [comment]
same

**Comment by stephenhawk8054** [comment]
We had many issues opened for spotify: https://github.com/uBlockOrigin/uAssets/issues?q=is%3Aissue+open.spotify.com

Maybe we should put a megathread for this site?

**Comment by dportvine** [comment]
> Maybe we should put a megathread for this site?

Also I see a lot of comments on [reddit.com](https://www.reddit.com/r/uBlockOrigin/comments/19fblme/no_longer_working_on_spotify_web/), so let's create one megathread.

**Comment by gettysburg** [comment]
I suggested a different approach to the Spotify Web Client ads taken from the desktop adblock "BlockTheSpot" linked here:

https://github.com/uBlockOrigin/uAssets/issues/22231

Essentially the desktop client is is a browser engine that downloads, executes and renders the same files from the same web-server as the web client runs on.

So, using their method, you can not only disable ad distribution completely, but also enable other features that usually only premium members get access to.

Unfortunately I was not aware of this issues existence.

**Comment by stephenhawk8054** [comment]
uBO blocking works by filters. Specific solutions for specific sites that need deeper modifications to the scripts require dedicated scripts for it, which can only be updated when the whole extension updated. It's better to leave those solutions to dedicated extensions.

**Comment by ItsProfessional** [comment]
> but also enable other features that usually only premium members get access to.

FYI: That's out of scope, we don't fix paywalls to features. Only the ads will be blocked.

**Comment by gettysburg** [comment]
@ItsProfessional That's fine.

@stephenhawk8054 Yeah, but does uBO not have the ability to inject scripts?

If so, could you inject something that changes the default state of a JS boolean or change some site specific code?

Take a look at the issue I posted to see what I mean, they essentially just wait for the browser engine to read a certain file and then patch it up.

**Comment by stephenhawk8054** [comment]
@gettysburg Scripts injections works by using a set of pre-defined scriptlets: https://github.com/gorhill/uBlock/wiki/Resources-Library

We don't write arbitrary long scripts for specific sites. As I said, that can only be maintained and updated when the whole extension updates. It cannot be updated frequently like filters whenever the site changes something.

**Comment by ItsProfessional** [comment]
@gettysburg There are hacky ways to inject scripts with `rpnt` and `replace`, but gorhill has been against the usage of such hacky techniques to inject code, as they are essentially RCEs. We want to avoid code manipulation (i.e. patches) to the sites scripts when possible.

**Comment by stephenhawk8054** [comment]
I see this request

![image](https://github.com/uBlockOrigin/uAssets/assets/66517106/ee5a892e-c987-444e-88b7-6f24faba010c)

I'll update

```adb
||2mdn.net/*/web_video_ads/$media,redirect=noop-1s.mp4:10,domain=open.spotify.com
```

**Comment by stephenhawk8054** [comment]
Anyone knows if using `*$media,redirect-rule=noop-1s.mp4:10,domain=open.spotify.com` caused any issues in the past?

**Comment by dportvine** [comment]
> I see this request
> 
<details>
<summary>Screen</summary>

>![image](https://private-user-images.githubusercontent.com/66517106/300237396-ee5a892e-c987-444e-88b7-6f24faba010c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY0NTc4MjAsIm5iZiI6MTcwNjQ1NzUyMCwicGF0aCI6Ii82NjUxNzEwNi8zMDAyMzczOTYtZWU1YTg5MmUtYzk4Ny00NDRlLTg4YjctNmYyNGZhYmEwMTBjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI4VDE1NTg0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ4MjVlNGYwZWZmMThjOTI2NWU2ZDQzZjRiMGNjZjE3Y2NiZjdlMzQwOWIwMDcyNjQxZjMxZDg5MDlkNDg5YmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.j06BgQe-n_580q4IbaWLMIQ_uozD7wRbXhygqgvGlqE)
</details>

>
> I'll update
> 
> ```
> ||2mdn.net^$media,redirect=noop-1s.mp4:10,domain=open.spotify.com
> ```

I have the same as before.

<details>
<summary>Screen</summary>

![2024-01-28_180325](https://github.com/uBlockOrigin/uAssets/assets/111344219/7dad9f3f-e270-416e-ad63-e5c58a305a8e)
</details>

**Comment by stephenhawk8054** [comment]
Yeah, the ads connections look quite random. Sometimes I got `scdn.co`, few times I see `2mdn.net`

**Comment by stephenhawk8054** [comment]
Can anyone test if this filter can stop any `media` ads from appearing?

```adb
open.spotify.com##+js(trusted-replace-fetch-response, /\{"metadata":\{"uri":"spotify:ad:.+?:"AD"\}/, , track-playback)
```

**Comment by Luminous-Journey** [comment]
As i mentioned in #22258, i didn't have any issues with visual adds showing up but the songs would start skipping, and whats really irritating is that it doesn't happen every time

**Comment by stephenhawk8054** [comment]
@Luminous-Journey Can you do these steps (follow strictly the step order, don't mix the steps orders, or skip any steps):

1. Click on uBO icon > &#9636; ["The logger" icon](https://github.com/gorhill/uBlock/wiki/The-logger), it will open a logger window
2. Open new tab and reproduce the issue
3. Switch to the logger window that was opened in step 1
4. Copy the logger via &#128203; [export/copy button at the top right](https://github.com/gorhill/uBlock/wiki/The-logger#export-dialog)
5. Paste the logs to https://www.logpasta.com

**Comment by Luminous-Journey** [comment]
Unfortunately, while i was attempting to troubleshoot, the bug fixed itself and now i cant record it until it happens again, and who knows when that will be, but when it does i will do as recommended

**Comment by Luminous-Journey** [comment]
Also @stephenhawk8054 do you want the markdown or plaintext version?

**Comment by stephenhawk8054** [comment]
@Luminous-Journey Any is fine.

**Comment by Luminous-Journey** [comment]
Another bug that i ran into with the same config is when a song is playing like an ad where you cant skip forward and it says advertisment in the bottom right, managed to record this one

log: https://www.logpasta.com/paste/91b347d8-8925-4e30-9fc8-eb17e5db82f5

screenshot: ![image](https://github.com/uBlockOrigin/uAssets/assets/77587691/0bb52a40-557f-410a-ab9e-f53f2b95f99b)

**Comment by stephenhawk8054** [comment]
Looks like you open the logger after loading the page. The right steps are opening the logger first and reproducing the issue at the website later (step 1 -> step 2)

**Comment by Luminous-Journey** [comment]
some more stuff got logged after a min or so 
https://www.logpasta.com/paste/94e5ae94-6fd5-4a56-afd8-3366cc89b71c

**Comment by Luminous-Journey** [comment]
I've had the logger open since you suggested recording the issue in case i ran into another issue

**Comment by stephenhawk8054** [comment]
Step 2 -> step 3 means you need to switch from the spotify page to the logger. Don't open any other sites and switch from that site to the logger

**Comment by Luminous-Journey** [comment]
So the logger only logs when its in the foreground?

**Comment by stephenhawk8054** [comment]
The logger needs to be opened first to start capturing information. Then depending on which tab you are opening, it will show the information from that tab. You can switch the tab in the logger's drop down.

-+-

@Luminous-Journey But anyways, can you do these steps:

- Go to `Settings` tab in uBO settings
- Scroll down and enable `I am an advanced user`, and click on the gear icon at the end
- Find the line `trustedListPrefixes` at the bottom and change the value from `ublock-` to `ublock- user-` (keep the space exactly) -> Click on `Apply changes`

Then can you test these filters?

    open.spotify.com##+js(trusted-replace-fetch-response, /\,\{"metadata":\{"uri":"spotify:ad:.+?:"AD"\}/, , track-playback)

Click on uBO icon > &#9881; [Dashboard button](https://github.com/gorhill/uBlock/wiki/Quick-guide:-popup-user-interface#the-tools) > Add the filter(s) in ["My filters" pane](https://github.com/gorhill/uBlock/wiki/Dashboard:-My-filters) > &#10003; Apply changes > Open new tab and test again.

Please also double-check again to see if the filters are marked as valid or invalid (it will show red line if invalid, otherwise it's good).

**Comment by Luminous-Journey** [comment]
The filter seems to be valid, I will post back here if something happens again. Tysm

**Comment by Luminous-Journey** [comment]
> The logger needs to be opened first to start capturing information. Then depending on which tab you are opening, it will show the information from that tab. You can switch the tab in the logger's drop down.

Thats good, I had the dropdown set to the spotify tab, and had opened the logger before reloading that page

**Comment by dportvine** [comment]
> Can anyone test if this filter can stop any `media` ads from appearing?
> 
> ```
> open.spotify.com##+js(trusted-replace-fetch-response, /\,\{"metadata":\{"uri":"spotify:ad:.+?:"AD"\}/, , track-playback)
> ```

This scriplet breaks the player on my side.

<details>
<summary>Screen</summary>

![2024-01-29_094811](https://github.com/uBlockOrigin/uAssets/assets/111344219/e93f6fb6-e6e8-4aad-bcb7-447e1f2dc840)
![2024-01-29_094853](https://github.com/uBlockOrigin/uAssets/assets/111344219/6d968ace-bc2a-43f2-b110-7297aa4db50c)
</details>

<details>
<summary>Video</summary>

https://github.com/uBlockOrigin/uAssets/assets/111344219/f125c8c0-88a7-40fa-b591-15e8344c870d
</details>

**Comment by stephenhawk8054** [comment]
@dportvine Keep the above filter, can you continue adding this?

```adb
open.spotify.com##+js(trusted-replace-fetch-response, "disallow_seeking":true, "disallow_seeking":false, track-playback)
```

**Comment by dportvine** [comment]
Didn't help.

**Comment by stephenhawk8054** [comment]
Hmm... I don't know how to resolve more. I can reproduce the issue, but cannot any more after adding that filter

**Comment by dportvine** [comment]
Nothing has changed for me.

<details>
<summary>Video</summary>

https://github.com/uBlockOrigin/uAssets/assets/111344219/d7b52094-a521-4438-8d25-0dc2eaf71b4d
</details>

**Comment by gettysburg** [comment]
> We want to avoid code manipulation (i.e. patches) to the site's scripts whenever possible, as this is basically RCE.

RCE, aka remote code execution, is not the correct term here.

The dynamic library from "BlockTheSpot" is patching `.css` files for example as they are parsed by the browser, in order to achieve a clean UI without the "Explore Premium" button for example, and the same is done with `.js` files by replacing or changing a JavaScript variable or changing existing JavaScript code, no new code is added and executed, simply existing code with known ad-provider URL's (for example `sentry.io`) in them is modified, in this case `sentry.io` is replaced by `localhost`.

This does not qualify as RCE whatsoever, as all changes are done locally and in the users favor, there is no harm, and no code execution that wouldn't occur otherwise.

Regardless, **this was just a suggestion**, if this is currently not possible as explained by @stephenhawk8054, then it's okay and the current approach of just playing 1 second noop files is good enough I guess, I managed to find an installer with which BlockTheSpot works, so I don't have to use the web player anymore:

<details><summary>My original ticket on BPS and the solution I found there:</summary>

[Ticket Link](https://github.com/mrpond/BlockTheSpot/issues/513#issuecomment-1914062706)

![image](https://github.com/uBlockOrigin/uAssets/assets/2256003/8979bca9-3eb7-41c6-9b27-18f0b28a971c)

</details>

Thank you regardless @ItsProfessional and @stephenhawk8054 for your hard work, and for your replies :-)

**Comment by stephenhawk8054** [comment]
@gettysburg RCE will be determined by the extension store reviewers. You can argue whatever you want, but the final decision is by them. Any potential argument would be a potential issue of wasting time for back-and-forth questions-explanations and potential for delayed/denied of future versions' publishing.

**Comment by gettysburg** [comment]
@stephenhawk8054 I have to admit that I did not consider Mozilla's review team into consideration, and the way that these new filters allowing for file modification could be abused by third party filter-list maintainers.

Thank you for bringing that up, I totally have not thought of that.

**Comment by gettysburg** [comment]
@stephenhawk8054 I have to admit that I did not take Mozilla's review team into consideration, and the way that these new filters allowing for file modification could be abused by (third party) filter-list maintainers.

Thank you for bringing that up, I totally have not thought of that.

**Comment by stephenhawk8054** [comment]
@dportvine This is a POC test, can you test if this skips the ads faster?

```adb
open.spotify.com##+js(trusted-replace-fetch-response, '/(\{"uri":"spotify:ad:.+?,"manifest"):.+?,"audio_id":/', '$1:{"file_urls_mp3":[{"bitrate":160000,"file_id":null,"file_url":"https://raw.githubusercontent.com/gorhill/uBlock/master/src/web_accessible_resources/noop-0.1s.mp3","impression_urls":[""],"track_type":"AUDIO","format":null,"audio_quality":null,"hifi_status":null,"gain_db":null,"expires_at":null}]},"audio_id":', track-playback)
```

**Comment by dportvine** [comment]
The player stopped. Реклама = Ads.

<details>
<summary>Screen</summary>

![2024-01-30_153428](https://github.com/uBlockOrigin/uAssets/assets/111344219/d7d6930c-cec1-4c99-9f0e-883070f40f26)
</details>

**Comment by dportvine** [comment]
I don't know what happened but that's gone. Scriplet works for me.

**Comment by dportvine** [comment]
logger

<details>
<summary>Screen</summary>

![2024-01-30_161047](https://github.com/uBlockOrigin/uAssets/assets/111344219/94731ffc-0df3-470a-8c4e-1bdfdc39c857)
</details>

**Comment by Luminous-Journey** [comment]
> As i mentioned in #22258, i didn't have any issues with visual adds showing up but the songs would start skipping, and whats really irritating is that it doesn't happen every time

I managed to get that one issue logged where the songs just skip instead of playing @stephenhawk8054 
https://www.logpasta.com/paste/c7f016dd-bd30-4c9a-977f-56f8f0a2af69

**Comment by Luminous-Journey** [comment]
But now it's fixed itself again...
I hate inconsistency when attempting to debug

**Comment by stephenhawk8054** [comment]
@Luminous-Journey Remove the custom filters I told you to test and check if your original issue is still there or not.

If I understand correctly, your original issue is the player was stuck and could not advance right?

**Comment by Luminous-Journey** [comment]
It would advance, but it would simply skip all the preceding songs instead of playing them

**Comment by stephenhawk8054** [comment]
@Luminous-Journey That's strange. For now just remove the current filters and see if the issue is still there or not. Also check your other extensions if possible too.

**Comment by Luminous-Journey** [comment]
The one you had me go to the advanced settings for?

**Comment by stephenhawk8054** [comment]
@Luminous-Journey Yeah, the ones I told you to add in `My filters`. I don't remember how many you are using now.

**Comment by Baysul** [comment]
![image](https://github.com/uBlockOrigin/uAssets/assets/7519920/d51d553d-95ff-40dc-8822-a86b2625daae)
The element is named `##.sponsor-container` or something along those lines, according to uBlock Origin's element picker. Seems somewhat inconsistent; I found this one in my Discover Weekly playlist but not in other playlists.

**Comment by bocharudo** [comment]
Hi, I'm using Spotify on Firefox Mobile. I have a free account, and whenever I try to search a song and play it, sometimes a 1 sec ad will show then Spotify randomly shuffles musics for me. Is there any way to bypass the random shffling for free accounts?

On my desktop, uBO works perfectly fine on playing specific songs. It's just on Firefox Mobile I got this issue.

**Comment by stephenhawk8054** [comment]
Backup your uBO configurations, reset uBO to default settings, restart the browser and test again.

**Comment by bocharudo** [comment]
Already did that multiple times. Unfortunately, for some reason, it won't really work.

**Comment by stephenhawk8054** [comment]
Can you share the troubleshooting information?

**Comment by bocharudo** [comment]
Firstly, I did reset uBO to its default settings. Then, I've restarted the Firefox app. Finally, open https://open.spotify.com then played a specific song. As expected, it still randomly shuffles songs. Did it dozen of times now lol. (I hope I'm answering your question right)

Is it kind of the Firefox mobile problem? because it works perfectly fine on my Firefox desktop.

**Comment by mapx-** [comment]
> then played a specific song. As expected, it still randomly shuffles songs

I don't understand: if you played a specific song what is shuffled ?

**Comment by bocharudo** [comment]
I apologize for the confusion, what I mean is when I tried to play a specific song, Spotify randomly plays a song aside from what I chose. I think it's because I use Spotify free version. What I'm expecting to happen is to bypass that randomizing of songs.

**Comment by mapx-** [comment]
@bocharudo if you turn off uBO does it still happen ?

**Comment by bocharudo** [comment]
yes it does

**Comment by mapx-** [comment]
If even disabling uBO you get the same behaviour, then it's nothing to do

**Comment by bocharudo** [comment]
I just read somewhere that Spotify web on desktop and Spotify desktop app allows free users to listen music without those shuffling.

But on mobile, either you use the android app or the web player, there is still that shuffling.

I just decided yesterday to go back on Spotify after using YouTube Music for such a long time. I think I'll discontinue that decision. YouTube Music allows me to select specific musics and I use it on web with uBO to block ads.

**Comment by stephenhawk8054** [comment]
@historymann Can you do these steps (follow strictly the step order, don't mix the steps orders, or skip any steps):

1. Click on uBO icon > ▤ ["The logger" icon](https://github.com/gorhill/uBlock/wiki/The-logger), it will open a logger window
2. Open new tab and reproduce the issue
3. Switch to the logger window that was opened in step 1
4. Copy the logger via 📋 [export/copy button at the top right](https://github.com/gorhill/uBlock/wiki/The-logger#export-dialog)
5. Paste the logs to https://www.logpasta.com and share the result here

**Comment by stephenhawk8054** [comment]
@xigzagoon Can you do the steps above: https://github.com/uBlockOrigin/uAssets/issues/22198#issuecomment-2673226622

**Comment by xigzagoon** [comment]
[spotify ad logger output.txt](https://github.com/user-attachments/files/19482519/spotify.ad.logger.output.txt)

Hopefully I did this right - it's probably super-long because I, surprisingly, couldn't get it to play an ad by switching between or skipping songs, I had to just leave music playing until I heard an ad. Two ads played in a row at about 3:35 and I paused a third as it started before pulling everything from the logger.

**Comment by stephenhawk8054** [comment]
@xigzagoon Can you test this filter:

```adb
||adstudio-assets.scdn.co^$media,redirect=noop-1s.mp4:10,domain=open.spotify.com
```

Click on uBO icon > ⚙ [Dashboard button](https://github.com/gorhill/uBlock/wiki/Quick-guide:-popup-user-interface#the-tools) > Add the filter(s) in ["My filters" pane](https://github.com/gorhill/uBlock/wiki/Dashboard:-My-filters) > ✓ Apply changes > Open new tab and test again.

**Comment by xigzagoon** [comment]
That seems to have done it!

**Comment by TheRealCreeperCats** [comment]
Hi! I hope that I'm doing this right since I keep seeing closed duplicate threads say to go to closed or currently open ones for help. 

A couple days ago I made the switch to Firefox (since Chrome completely killed uBO for me) and as of the past couple days uBO hasn't been fully skipping or moving past the audio ads on Spotify. It'll still "skip" them/shorten them down and the bar will progress, but once it does it just stays there and doesn't move onto the next ad or song. I can't skip the ad, pause it, or move the progress bar. If I reload the page it kicks me off of the playlist and I need to replay it again. I've tried updating the filters, removing and re-adding uBO to Firefox, and other things I saw recommended in other solved threads, but nothing has worked so far. I hope I'm able to get some help here and to hear back soon! Thank you!

**Comment by stephenhawk8054** [comment]
Can you do these steps: https://github.com/uBlockOrigin/uAssets/issues/22198#issuecomment-2673226622

**Comment by TheRealCreeperCats** [comment]
Yep! Here you go: https://www.logpasta.com/paste/dfc8c5ea-7bd3-4ae9-9d62-42766c5253f9

**Comment by stephenhawk8054** [comment]
I can't find abnormal things in the log so I can't investigate further.

**Comment by TheRealCreeperCats** [comment]
Well dang, I guess I'm outta luck with this after all :(

**Comment by stephenhawk8054** [comment]
@TheRealCreeperCats Can you test these filters:

```adb
open.spotify.com#@#+js(trusted-json-edit-fetch-response, '.cdnurl=["data:audio/mpeg;base64,SUQzBAAAAAAAI1RTU0UAAAAPAAADTGF2ZjU2LjQwLjEwMQAAAAAAAAAAAAAA//tUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAASW5mbwAAAA8AAAAGAAADAABgYGBgYGBgYGBgYGBgYGBggICAgICAgICAgICAgICAgICgoKCgoKCgoKCgoKCgoKCgwMDAwMDAwMDAwMDAwMDAwMDg4ODg4ODg4ODg4ODg4ODg4P////////////////////8AAAAATGF2YzU2LjYwAAAAAAAAAAAAAAAAJAAAAAAAAAAAAwDNZKlY//sUZAAP8AAAaQAAAAgAAA0gAAABAAABpAAAACAAADSAAAAETEFNRTMuOTkuNVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVV//sUZB4P8AAAaQAAAAgAAA0gAAABAAABpAAAACAAADSAAAAEVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVV//sUZDwP8AAAaQAAAAgAAA0gAAABAAABpAAAACAAADSAAAAEVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVV//sUZFoP8AAAaQAAAAgAAA0gAAABAAABpAAAACAAADSAAAAEVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVV//sUZHgP8AAAaQAAAAgAAA0gAAABAAABpAAAACAAADSAAAAEVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVV//sUZJYP8AAAaQAAAAgAAA0gAAABAAABpAAAACAAADSAAAAEVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVV"]', propsToMatch, /storage-resolve/files/audio/interactive)
```

Click on uBO icon > ⚙ [Dashboard button](https://github.com/gorhill/uBlock/wiki/Quick-guide:-popup-user-interface#the-tools) > Add the filter(s) in ["My filters" pane](https://github.com/gorhill/uBlock/wiki/Dashboard:-My-filters) > **Turn on `Allow custom filters requiring trust`** > ✓ Apply changes > Open new tab and test again.

**Comment by pamvalle** [comment]
Everytime a Google Advertisement is showed, it won't skipped or play another song until we reload the site again.

**Comment by stephenhawk8054** [comment]
@pamvalle Can you test the above filters?

**Comment by TheRealCreeperCats** [comment]
The filters seem to work! They'll still sit on the ads for a varying amounts of time, so It takes about 20-50 seconds (average is around 30-35 seconds it seems) to get through them all and move on, but it does eventually move on and continue with the playlist! It seems some ads will take longer than others. But at the end of the day it does eventually move past them.

**Comment by stephenhawk8054** [comment]
> They'll still sit on the ads for a varying amounts of time, so It takes about 20-50 seconds (average is around 30-35 seconds it seems) to get through them all and move on

I don't understand what this means. Can you record video how it looks like?

Also, can you screenshot "My filters" pane?

**Comment by stephenhawk8054** [comment]
@TheRealCreeperCats Can you remove the 2nd filter and test again?

**Comment by gettysburg** [comment]
Usually I use the Spotify desktop application for Linux on my ArchLinux workstation (together with [SpotX](https://github.com/SpotX-Official/SpotX-Bash)), but the few times I used the browser player, the described behavior by @TheRealCreeperCats seems to be fixed almost completely, and I can also clarify on what he meant by "sitting on the ads".

Before, when there was one or more audio ads, Spotify would, as per the old uBlock Origin rules, play the `noop.mp3` file as intended, however their ad delivery system still insisted on waiting at least 10 seconds before starting playback of music.

Now, with the new rules, the ads are skipped way faster, there is still a minimal wait period but it is very acceptable (around 1-3 seconds last time I checked).

@stephenhawk8054: Thank you for the good work and continuing to support the Spotify filters.

**Comment by theblackkaizer** [comment]
https://www.logpasta.com/paste/d776e02b-262d-4534-af09-fb5df84112a1

am running into same issues. log above. for reference, this time it bypassed the first wave of ads, I continued force skipping songs until the second wave which froze play.

I am using this from in My Filters found in the July 13th reply.

```adb
open.spotify.com#@#+js(trusted-json-edit-fetch-response, '.cdnurl=["data:audio/mpeg;base64,SUQzBAAAAAAAI1RTU0UAAAAPAAADTGF2ZjU2LjQwLjEwMQAAAAAAAAAAAAAA//tUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAASW5mbwAAAA8AAAAGAAADAABgYGBgYGBgYGBgYGBgYGBggICAgICAgICAgICAgICAgICgoKCgoKCgoKCgoKCgoKCgwMDAwMDAwMDAwMDAwMDAwMDg4ODg4ODg4ODg4ODg4ODg4P////////////////////8AAAAATGF2YzU2LjYwAAAAAAAAAAAAAAAAJAAAAAAAAAAAAwDNZKlY//sUZAAP8AAAaQAAAAgAAA0gAAABAAABpAAAACAAADSAAAAETEFNRTMuOTkuNVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVV//sUZB4P8AAAaQAAAAgAAA0gAAABAAABpAAAACAAADSAAAAEVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVV//sUZDwP8AAAaQAAAAgAAA0gAAABAAABpAAAACAAADSAAAAEVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVV//sUZFoP8AAAaQAAAAgAAA0gAAABAAABpAAAACAAADSAAAAEVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVV//sUZHgP8AAAaQAAAAgAAA0gAAABAAABpAAAACAAADSAAAAEVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVV//sUZJYP8AAAaQAAAAgAAA0gAAABAAABpAAAACAAADSAAAAEVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVV"]', propsToMatch, /storage-resolve/files/audio/interactive)
```

**Comment by 21abdela** [comment]
so for some reason , it skips the first 5 songs, flashing the message 'spotify cant play this rn' and plays the first 10 seconds of the sixth song, and goes silent after 10 seconds

**Comment by stephenhawk8054** [comment]
**Chromium uBOL**

Redirecting on MV3 still has issues: https://github.com/uBlockOrigin/uBOL-home/issues/293#issuecomment-2701258312

**Comment by mikkorantalainen** [comment]
I'm using uBlock Origin Lite with latest Chrome and spotify seems to stop when an ad should play. That is, I will not hear ad but the audio playback of the actual content will not resume either.

**Comment by stephenhawk8054** [comment]
https://github.com/uBlockOrigin/uAssets/issues/22198#issuecomment-3821876950

**Comment by solomoncyj** [comment]
IDK if it belong here, but i have an intermittent  problem where the media controls will freeze and become unclickable, and will only be unfrozen by unloading the ad blocker or pausing via system media controls. firefox on linux

**Comment by stephenhawk8054** [comment]
Test again in a new Firefox profile with only uBO and its default settings.

**Comment by Cederick** [comment]
I started to have the same issue today on PC.

It skips some songs and then plays the first 10 seconds of a song before going silent. It also stops completely after this song is done. I have not added any new filter list lately.

**Comment by Cederick** [comment]
Small update. Reinstalling/updating Firefox Nightly by downloading the latest version and isntalling it fixed it for me for now.

**Comment by DesperatePea** [comment]
There seems to be a problem with safari 26 and uboL on spotify for me at the moment. I tried opening open.spotify.com on a private window, i tried resetting uboL to it's default state, waited 30 seconds after restarting safari, waited for filters to load, but i still get the same error.

Spotify says "error occoured, reload to continue" and it cannot play albums. Occasionally plays a random track but if i tap play, the error occours. 

I verified this is a UBOL issue as this does not happen if i turn it off. 

<details>

```yaml
name: uBlock Origin Lite
version: 2026.222.1657
browser: Safari Mobile 26.3 (ios)
filtering:
 site: complete
 default: complete
permission: all
badge: 6
rulesets:
 +annoyances-ai
 +annoyances-cookies
 +annoyances-notifications
 +annoyances-others
 +annoyances-overlays
 +annoyances-social
 +annoyances-widgets
 +ublock-experimental
 +ubol-tests
 adguard-mobile
 easylist
 easyprivacy
 nor-0
 pgl
 ublock-badware
 ublock-filters
scripting:
 ublock-badware.main
 ubol-tests.main
 ublock-experimental.main
 adguard-mobile.main
 adguard-mobile.isolated
 css-generic-some
 annoyances-social.isolated
 ublock-filters.isolated
 annoyances-cookies.main
 css-generic-all
 annoyances-cookies.isolated
 annoyances-notifications.main
 easyprivacy.main
 css-generichigh
 annoyances-overlays.main
 annoyances-others.main
 easyprivacy.isolated
 annoyances-others.isolated
 nor-0.main
 annoyances-overlays.isolated
 ublock-experimental.isolated
 css-specific
 ublock-filters.main
 annoyances-notifications.isolated
 css-procedural
```
</details>

Screenshot:
https://github.com/user-attachments/assets/34cd1ebe-f4fd-4204-9e7d-e120a68e74d1" />

**Comment by lunaflight** [comment]
# copy pasteability
this is useful when the song title or artist is something u want to copy paste and send your friends or something especially when typing it is hard. 

## nuisance
i feel that the song title and artist here should be copy-pasteable.

https://github.com/user-attachments/assets/96f1f0c5-edb5-47a3-9d1c-3327725cfcc2" />

it should at least be copyable if your mouse starts from the red line and drags over, since nothing happens anyway,
but if your mouse is in the blue region, i think there is some functionality where spotify lets u drag it into a playlist, so this should be preserved?

https://github.com/user-attachments/assets/4ea386eb-de6e-40ca-af18-f0b7b2650a6d" />

## steps to reprod
1. open.spotify.com
2. play any song
3. bottom bar's title n artist is not copy pasteable

## nuisance 2
i think that the song titles and everything here should also be copy-pasteable

again, it should be copypasteable only if the mouse starts from the red line and drags across like that.
if u try to start from the middle, there is some custom drag functionality that should be preserved.

https://github.com/user-attachments/assets/659739db-2d88-4cda-afd0-e7bcc263b8b1" />

## steps to reprod
1. visit https://open.spotify.com/album/2wQqd6G4SLmevXyGwoQbuZ or any playlist
2. try selecting as described above

## sidebar

i think the stuff in the sidebar in the home page open.spotify.com should also be copy pasteable in the same idea

https://github.com/user-attachments/assets/a6df2e30-46d3-406e-a73e-750fcaa47769" />

## homepage

on open.spotify.com , all of these should also be copy pasteable if selecting from outside of the draggable region

https://github.com/user-attachments/assets/6d064ca2-a595-4738-a451-c9afc97e96ea" />

## search bar

same, i can drag from the red line without triggering any special functionality

https://github.com/user-attachments/assets/6abbc263-5e81-4338-9905-f6b10d411873" />

## right sidebar

honestly like everything here should be copy pasteable. the highlighting is what can actually be copy pasted, since i am trying to select everything

this is the right sidebar when you are listening to a song

https://github.com/user-attachments/assets/b56932f4-ce47-455e-8a71-8e00be60c0b3" />

## credits

when playing a song, in the right sidebar, click show all in the credits block. the popup contains non copypasteable stuff too

https://github.com/user-attachments/assets/c66d32ad-a3ea-46c8-bde4-543faa5f78dd" />

**Comment by mapx-** [comment]
@lunaflight test

```adb
open.spotify.com##+js(aeld, contextmenu)
open.spotify.com##*:style(-webkit-touch-callout: default !important; -webkit-user-select: text !important; -khtml-user-select: text !important; -moz-user-select: text !important; -ms-user-select: text !important; user-select: text !important; -webkit-tap-highlight-color: revert !important; cursor: revert !important;)
```

**Comment by stephenhawk8054** [comment]
`open.spotify.com##+js(aeld, contextmenu)` will break spotify's own context menu. I don't think we should use that in stock lists.

**Comment by lunaflight** [comment]
just 

```
open.spotify.com##*:style(-webkit-touch-callout: default !important; -webkit-user-select: text !important; -khtml-user-select: text !important; -moz-user-select: text !important; -ms-user-select: text !important; user-select: text !important; -webkit-tap-highlight-color: revert !important; cursor: revert !important;)
```

alone seems to do the trick!
- it resolves all copy paste problems
- right click menus all work
- dragging songs from various places (e.g. search bar or bottom _now playing_ bar) into other places (e.g. playlists) all work and look exactly like how it works without the filter on

**Comment by lunaflight** [comment]
hiya, upon some usage with the filter, i think we should remove `cursor: revert !important`, i.e. use this filter:

```
open.spotify.com##*:style(-webkit-touch-callout: default !important; -webkit-user-select: text !important; -khtml-user-select: text !important; -moz-user-select: text !important; -ms-user-select: text !important; user-select: text !important; -webkit-tap-highlight-color: revert !important;)
```

1) spotify doesn't use any crazy cursor in the first place
2) reverting the cursor actually causes some cosmetic unexpectedness, for example, when hovering your cursor over the pause and play button on the bottom, a reverted cursor will only show a default cursor, when it should be showing the cursor that implies that it is clickable.
3) all functionality is still preserved as per the above discussion

i'm using Firefox and GNOME for what it's worth.

**Comment by dijonkitchen** [comment]
Unable to play album on Safari with default uBO lite. 

Example: https://open.spotify.com/album/7EJ0OT5ZqybXxcYRa6mccM

```
name: uBlock Origin Lite
version: 2026.322.1735
browser: Safari 26.3 (mac)
filtering:
 default: optimal
permission: all
badge: ?
rules: 17809
rulesets:
 easylist
 easyprivacy
 ublock-filters
scripting:
 css-specific
 ublock-filters.isolated
 ublock-filters.main
 toolbar-icon
 easyprivacy.main
 css-procedural
 easyprivacy.isolated
```

**Comment by zapf740** [comment]
Spotify skips songs automatically when ublock is activiated

**Comment by DimitrisKalaboukas** [comment]
- uBlock filters – Ads
```
||spotifycdn.com/audio/$media,redirect=noop-1s.mp4:10,domain=open.spotify.com
```

causes breakage of Spotify Web Player on Android, e.g. on:

`https://open.spotify.com/album/1bEgfrSNnaUK4KGEsFRBo7`

tapping play button does nothing.

<details>

<img width="1080" height="2340" alt="Image" src="https://github.com/user-attachments/assets/a10b489f-3b0e-4303-ab01-b31832a8ccc3" />

</details>

**Comment by DimitrisKalaboukas** [comment]
@stephenhawk8054 The breakage I mentioned above affects MV2 uBO on Android, e.g. in Microsoft Edge.

**Comment by stephenhawk8054** [comment]
@DimitrisKalaboukas Yes, the update is not for your issue. I can't reproduce your issue. Test again in official Firefox browser.

**Comment by DimitrisKalaboukas** [comment]
@stephenhawk8054 Then why the exclusion filter:

```
@@||spotifycdn.com/audio/$media,redirect=noop-1s.mp4:10,domain=open.spotify.com
```

solves the issue? Is it a conflict with AdGuard Base Filter or EasyList?

[uBlock Origin settings.txt](https://github.com/user-attachments/files/27235673/uBlock.Origin.settings.txt)

**Comment by stephenhawk8054** [comment]
I don't know. I said

> I can't reproduce your issue.

I also said

> Test again in official Firefox browser.

**Comment by ugfcxfy790c** [comment]
Spotify just breaks completely for me when I add the uBO Lite extension (Safari) and refuses to play anything. When I turn off uBlock on that site, it works fine. It breaks even in basic filtering mode.

**Comment by stephenhawk8054** [comment]
Can you screen record how the issue looks like?

**Comment by 4raft** [comment]
> Can you screen record how the issue looks like?

I'm also experiencing the same issue, here's a screen recording showcasing the issue (first with uBlock turned off, then with it turned on for the issue), saying 'Spotify can't play this right now'

https://files.catbox.moe/ojfyfc.mov

Also I looked at the console in Safari, and found the following 2 errors popping up every time I press the play button with uBO Lite turned on

https://files.catbox.moe/u2xgqo.png


```
[Error] Failed to load resource: the server responded with a status of 404 () (749c8db164e25762e24f9758e07653f7a99f79f0, line 0)

[Error] Unhandled Promise Rejection: PlayerAPIClientError: No such device.
``` 
<details><summary>Details</summary>
<p>

name: uBlock Origin Lite
version: 2026.621.1813
browser: Safari 27.0 (mac)
filtering:
 site: basic
 default: optimal
permission: all
badge: 9
rules: 115619
rulesets:
 easylist
 easyprivacy
 ublock-filters
scripting:
 easyprivacy.main
 css-specific
 ublock-filters.main
 prevent-popup
 easyprivacy.isolated
 toolbar-icon
 ublock-filters.isolated

</p>
</details>

**Comment by stephenhawk8054** [comment]
What happens if you set to Optimal mode?

**Comment by 4raft** [comment]
Same thing happens in both Optimal and Complete mode, with the same messages in the logs as well.

**Comment by gettysburg** [comment]
Slightly off-topic:

I do believe that, at least using the Spotify web app, that they do sometimes "shadow-ban" certain users if they are able to detect that you are trying to block their ads.

I don't have concrete proof of this, but 1-2 years ago, when using uBlock Origin with all filter lists, Spotify stopped working for me, even when using the Desktop app, and it would show me an error message (genuine Spotify Desktop client, no patches against ads).

Changing my IP address, clearing all related Spotify website data and creating a new account fixed it for a while.

Make of that what you will.

Recently, with the latest uBlock Origin filter lists, I never had any problems, and I now use a patched Spotify Desktop client to block the ads there too.
